File: 1207.unique-number-of-occurrences.cpp
/*
 * @lc app=leetcode id=1207 lang=cpp
 *
 * [1207] Unique Number of Occurrences
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    bool uniqueOccurrences(vector<int> &arr)
    {
        vector<int> freq(2001);
        for (auto n : arr)
        {
            ++freq[n + 1000];
        }
        vector<bool> seen_freq(1001);
        for (auto n : freq)
        {
            if (n != 0)
            {
                if (seen_freq[n])
                {
                    return false;
                }
                else
                {
                    seen_freq[n] = true;
                }
            }
        }
        return true;
    }
};
// @lc code=end

File: java/1207.unique-number-of-occurrences.java
/*
 * @lc app=leetcode id=1207 lang=java
 *
 * [1207] Unique Number of Occurrences
 */

// @lc code=start
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        int[] freq = new int[2001];
        for (var n : arr) {
            ++freq[1000 + n];
        }
        boolean[] seen_freq = new boolean[1001];
        for (var n : freq) {
            if (n != 0) {
                if (seen_freq[n]) {
                    return false;
                } else {
                    seen_freq[n] = true;
                }
            }
        }
        return true;
    }
}
// @lc code=end
